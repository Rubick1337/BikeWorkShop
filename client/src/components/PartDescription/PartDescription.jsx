import React from 'react';

const PartDescription = ({ part, getCategoryName, getTypeName }) => {
    const apiBaseUrl = process.env.REACT_APP_API_BASE_URL;

    return (
        <div className="modal__information">
            <img
                src={`${apiBaseUrl}${part.img}`}
                alt={part.name}
                style={{ width: '250px', marginBottom: '20px',alignSelf: 'center'}}
            />
            <p><strong>Тип детали:</strong> {getTypeName(part.id_type_part)}</p>
            <p><strong>Категория товара:</strong> {getCategoryName(part.id_category_part)}</p>
            <p><strong>Брэнд:</strong> {part.brand}</p>
            <p><strong>Модель:</strong> {part.model}</p>
            <p><strong>Цена:</strong> ${part.price}</p>
            <p><strong>Описание:</strong> {part.description}</p>
        </div>
    );
};

export default PartDescription;
