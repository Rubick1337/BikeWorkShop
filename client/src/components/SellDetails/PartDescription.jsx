import React from 'react';

const PartDescription = ({ part, getCategoryName, getTypeName }) => {
    return (
        <div className="modal__information">
            <img
                src={`http://localhost:9005/${part.img}`}
                alt={part.name}
                style={{ width: '250px', marginBottom: '20px',alignSelf: 'center'}}
            />
            <p><strong>Тип детали:</strong> {getTypeName(part.id_type_part)}</p>
            <p><strong>Категория товара:</strong> {getCategoryName(part.id_category_part)}</p>
            <p><strong>Брэнд:</strong> {part.brand}</p>
            <p><strong>Модель:</strong> {part.model}</p>
            <p><strong>Цена:</strong> ${part.price}</p>
            <p><strong>Описание:</strong> {part.description}</p>
        </div>
    );
};

export default PartDescription;
