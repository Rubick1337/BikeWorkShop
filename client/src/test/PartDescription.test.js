import React from 'react';
import { render } from '@testing-library/react';
import PartDescription from '../components/PartDescription/PartDescription';

describe('PartDescription Component', () => {
    const mockPart = {
        id_type_part: '1',
        id_category_part: '2',
        img: '/path/to/image.jpg',
        brand: 'BrandName',
        model: 'ModelName',
        price: 100,
        description: 'This is a test part description.'
    };

    const mockGetTypeName = jest.fn(() => 'Тип детали');
    const mockGetCategoryName = jest.fn(() => 'Категория товара');

    it('matches snapshot', () => {
        const { asFragment } = render(
            <PartDescription
                part={mockPart}
                getTypeName={mockGetTypeName}
                getCategoryName={mockGetCategoryName}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});