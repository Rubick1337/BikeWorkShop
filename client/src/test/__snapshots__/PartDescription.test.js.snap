// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartDescription Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal__information"
  >
    <img
      src="http://localhost:9005//path/to/image.jpg"
      style="width: 250px; margin-bottom: 20px; align-self: center;"
    />
    <p>
      <strong>
        Тип детали:
      </strong>
       
    </p>
    <p>
      <strong>
        Категория товара:
      </strong>
       
    </p>
    <p>
      <strong>
        Брэнд:
      </strong>
       BrandName
    </p>
    <p>
      <strong>
        Модель:
      </strong>
       ModelName
    </p>
    <p>
      <strong>
        Цена:
      </strong>
       $100
    </p>
    <p>
      <strong>
        Описание:
      </strong>
       This is a test part description.
    </p>
  </div>
</DocumentFragment>
`;
